client/src/App.vue

```diff
@@ -0,0 +1,31 @@
+<script setup lang="ts">
+import { RouterLink, RouterView } from 'vue-router'
+import { onMounted } from 'vue';
+import { states } from './game/states'
+import Socket from './game/socket'
+
+onMounted(() => {
+    // connect when mounted and only when socket wasn't created
+    if (!states.gameSocket) {
```

Comment:
dont use `!` since its unclear what ur trying to do, use `== null` or something instead